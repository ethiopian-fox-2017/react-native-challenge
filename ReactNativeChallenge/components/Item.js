import React from 'react';
import {
  View,
  Text,
  Image
} from 'react-native';

const styles = {
  itemStyle: {
    flexDirection: 'column'
  },
  imageStyle: {
    width: 300,
    height: 200
  }
}

const Item = (props) => {
  return (
    <View style={styles.itemStyle}>
      
      <Text>{props.itemId} | {props.itemName} </Text>
    </View>
  )
}

export default Item;
