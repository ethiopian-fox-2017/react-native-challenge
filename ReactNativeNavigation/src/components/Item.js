import React from 'react';
import {
  View,
  Text,
  Image,
  TouchableHighlight,
  Dimensions
} from 'react-native';

const deviceWidth = Dimensions.get('window').width;
const deviceHeight = Dimensions.get('window').height;

const styles = {

  imageStyle: {
    width: deviceWidth * 0.327,
    height: deviceWidth * 0.327,
    position: 'relative'
  }
}

const Item = (props) => {
  const { navigate } = props.navigation;
  return (
    <View >
      <TouchableHighlight onPress={() => navigate('Detail', { item: props.item }) }>
        <Image style={styles.imageStyle}
        source={{uri: props.item.image_url[0]}}
        />
      </TouchableHighlight>

    </View>
  )
}

export default Item;
