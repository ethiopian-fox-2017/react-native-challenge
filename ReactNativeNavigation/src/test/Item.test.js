import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';

import Item from '../components/Item';


it('render and match to snapshot', () => {
  const dummyItem = {
    image_url: ''
  }
  const navigation = {
    navigate: function(name){

    }
  }
  const tree = renderer.create(
      <Item item={dummyItem} navigation={navigation}/>
  )
  expect(tree).toMatchSnapshot();
})
