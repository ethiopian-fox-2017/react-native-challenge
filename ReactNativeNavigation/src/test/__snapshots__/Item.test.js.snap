// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render and match to snapshot 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "height": 245.25,
          "position": "relative",
          "width": 245.25,
        }
      }
    />
  </View>
</View>
`;
